src/CTA.Rules.Metrics/PortSolutionResultReportGenerator.cs

```diff
@@ -45,20 +48,32 @@ private void GenerateMetrics()
             DownloadedFilesMetrics = MetricsTransformer.TransformDownloadedFiles(Context, PortSolutionResult.DownloadedFiles);
 
             // Gather project-level metrics
+            var targetVersionMetrics = new List<TargetVersionMetric>();
+            var upgradePackageMetrics = new List<UpgradePackageMetric>();
+            var actionPackageMetrics = new List<ActionPackageMetric>();
             var actionExecutionMetrics = new List<GenericActionExecutionMetric>();
+
             foreach (var projectResult in PortSolutionResult.ProjectResults)
             {
-                var executedActionMetrics = MetricsTransformer.TransformGenericActionExecutions(Context, projectResult);
-                actionExecutionMetrics.AddRange(executedActionMetrics);
+                targetVersionMetrics.AddRange(MetricsTransformer.TransformTargetVersions(Context, projectResult));
+                upgradePackageMetrics.AddRange(MetricsTransformer.TransformUpgradePackages(Context, projectResult));
+                actionPackageMetrics.AddRange(MetricsTransformer.TransformActionPackages(Context, projectResult));
+                actionExecutionMetrics.AddRange(MetricsTransformer.TransformGenericActionExecutions(Context, projectResult));
```

Comment:
if these methods are time consuming you can make them async.. if not time consuming ok to keep in this way...